src/functions-and-arrays.js

```diff
@@ -1,73 +1,597 @@
 // Iteration #1: Find the maximum
+function maxOfTwoNumbers(num1, num2) {
+  if (num1 > num2) {
+    return num1;
+  } else {
+    return num2;
+  }
+}
 
 // Iteration #2: Find longest word
-const words = ['mystery', 'brother', 'aviator', 'crocodile', 'pearl', 'orchard', 'crackpot'];
+const words = [
+  "mystery",
+  "brother",
+  "aviator",
+  "crocodile",
+  "pearl",
+  "orchard",
+  "crackpot",
+];
+
+function findLongestWord(array) {
+  let longestWord = "";
+  if (!array.length) {
+    return null;
+  }
+
+  for (let i = 0; i < array.length; i++) {
+    if (longestWord.length < array[i].length) {
+      longestWord = array[i];
+    } else {
+      continue;
+    }
+  }
+  return longestWord;
+}
 
 // Iteration #3: Calculate the sum
 
 const numbers = [6, 12, 1, 18, 13, 16, 2, 1, 8, 10];
 
+function sumNumbers(array) {
+  let sum = 0;
+
+  for (let i = 0; i < array.length; i++) {
+    if (typeof array[i] != "number") {
+      continue;
+    } else {
+      sum = sum + array[i];
+    }
+  }
+  return sum;
+}
+
+function sum(array) {
+  let sum = 0;
+
+  for (let i = 0; i < array.length; i++) {
+    if (
+      !(
+        typeof array[i] === "number" ||
+        typeof array[i] === "string" ||
+        typeof array[i] === "boolean"
+      )
+    ) {
+      throw new Error("Unsupported data type sir or ma'am");
+    } else if (typeof array[i] === "number") {
+      sum = sum + array[i];
+    } else if (typeof array[i] === "string") {
+      sum = sum + array[i].length;
+    } else if (typeof array[i] === "boolean") {
+      if (array[i] === true) {
+        sum++;
+      } else {
+        continue;
+      }
+    }
+  }
+  return sum;
+}
+
 // Iteration #4: Calculate the average
 // Level 1: Array of numbers
 const numbersAvg = [2, 6, 9, 10, 7, 4, 1, 9];
 
+function averageNumbers(array) {
+  if (!array.length) {
+    return null;
+  }
+  let sum = sumNumbers(array);
+  let average = sum / array.length;
+  return average;
```

Comment:
I feel it would also work if you had `return sumNumbers(array)/array.length`, you usually want to use variables when you plan to pass a value to another process/construct in your code but if that's not the case you can just return it